lib/trie.ex

```diff
@@ -87,6 +87,7 @@ defmodule MerklePatriciaTree.Trie do
 
     case Node.decode_trie(trie) do
       :empty -> nil # no node, bail
+      :not_found -> :not_found
```

Comment:
So, this is going to cause issues since the result is now going to be `binary() | :not_found | nil` which isn't great to pattern match against. I'd personally suggest we add a second function `fetch` that returns `{:ok, binary} | :not_found` and keep `get` as a variety that returns a `binary | nil`. Just my thoughts on keeping the interface clean.

Reply:
Good call.  I just made `Node#decode_trie` return `:empty` instead of `:not_found`. That removes the `:not_found` case here so this will return only return `binary() | nil` as expected